### PR TITLE
include POH in diagnostic APIs like walk_survivors.

### PR DESCRIPTION
**Pull request by VSadov**
Follow-up item for: https://github.com/dotnet/runtime/issues/13733

Reporting POH survivors to diagnostic APIs.

### CODE REVIEW

**Comment by VSadov** [comment]
I think this is ready for review.

File: src/coreclr/src/vm/gcenv.ee.cpp [hunk 1]
```diff
@@ -763,14 +763,15 @@ void GCToEEInterface::DiagWalkSurvivors(void* gcContext, bool fCompacting)
...
-void GCToEEInterface::DiagWalkLOHSurvivors(void* gcContext)
+void GCToEEInterface::DiagWalkUOHSurvivors(void* gcContext, int gen)
 {
 #if defined(GC_PROFILING) || defined(FEATURE_EVENT_TRACE)
     if (ShouldTrackSurvivorsForProfilerOrEtw())
     {
         size_t context = 0;
         ETW::GCLog::BeginMovedReferences(&context);
-        GCHeapUtilities::GetGCHeap()->DiagWalkSurvivorsWithType(gcContext, &WalkMovedReferences, (void*)context, walk_for_loh);
+        walk_surv_type type = (gen == 3) ? walk_for_loh : walk_for_poh;
```
Comment (Maoni0) [inline src/coreclr/src/vm/gcenv.ee.cpp:773]: > (gen == 3) [](start = 29, length = 11)

it seems very odd to have a hard coded value here. 

you could just have DiagWalkSurvivorsWithType take an additional parameter which is the generation.

**Review by Maoni0 — CHANGES_REQUESTED** [review]
the rest LGTM

File: src/coreclr/src/vm/gcenv.ee.cpp (hunk 1 above)
Comment (VSadov) [reply src/coreclr/src/vm/gcenv.ee.cpp:773]: I probably wanted to minimize changes. Using literal indeed looks strange now. Will change.

**Review by Maoni0 — APPROVED** [review]
LGTM!

**Comment by VSadov** [comment]
Thanks!!!

**Pull request merged**